### PR TITLE
Add a smarty parser.

### PR DESCRIPTION
**Pull request by jashook**
This can be used in the future by the arm64 ci to report output.

@kyulee1 @briansull PTAL

/cc @RussKeldorph

### CODE REVIEW

File: tests/scripts/smarty_parser.py [hunk 1]
```diff
@@ -0,0 +1,202 @@
+################################################################################
```
Comment (RussKeldorph) [inline tests/scripts/smarty_parser.py]: License?

Comment (jashook) [inline tests/scripts/smarty_parser.py]: Addressed

**Comment by kyulee1** [comment]
LGTM

**Comment by briansull** [comment]
LGTM

**Pull request merged**